src/analyses/ai.h

```diff
@@ -457,6 +513,12 @@ class ait:public ai_baset
   }
 };
 
+/// Base class for concurrency-aware static analyses. See
+/// \ref ait for details.
+/// The only difference is that after the sequential fixed point construction,
+/// as done by \ref ait, another step is added to account for
+/// concurrently-executed instructions.
+/// TODO: Somebody who truly understands this algorithm should document it!
```

Comment:
It's basically "Analysis of Multithreaded Programs" by Martin Rinard, SAS 2001. And that is basically making the analysis flow-insensitive by feeding results of a sequential execution back into the entry point, thereby accounting for any values computed by different threads.

Reply:
I took the liberty of copying your description and linking to the paper.